addons/addon-base-raas/packages/base-raas-services/lib/plugins/env-provisioning-plugin.js

```diff
@@ -211,7 +211,7 @@ async function updateEnvOnTerminationSuccess({ requestContext, container, status
     status,
     inWorkflow: 'false',
   };
-  const updatedEnvironment = await environmentScService.update(requestContext, environment);
+  const updatedEnvironment = await environmentScService.update(requestContext, environment, { action: 'REMOVE' });
```

Comment:
Just for my knowledge- is this supposed to run for just RStudio on stop as well? The filename seemed like this was pretty generic, so can you explain where this fits in. Also wondering why isn't there a rstudio-stop js file where we can add this?

Reply:
This plugin method is only used for environment termination, since the environment termination workflow uses this plugin to handle all actions after termination success, this is where it fits in. rstudio-stop is reusing ec2-stop js file, so no extra update is needed for rsdutio-sop.

Reply:
One clarifying question- Is this redundant piece of code if ec2-stop already has the ip removal code?

Reply:
Yes! The environment DB record update can be reused by all the workflow with some small twist in pass in parameters, and use plugin to invoke the method. I'll create a tech debt ticket to centralize the DB record update part. The ip removal code is essentially pass in one extra parameter when calling the environmentScService, so it shouldn't add up to existing tech debt.

Reply:
Actually we already have a ticket for it - https://sim.amazon.com/issues/GALI-491